### PR TITLE
Small fixes for cupy.sort

### PR DESCRIPTION
**Pull request by takagi**
Follows #186. This PR is a set of small fixes for `cupy.sort`.

### CODE REVIEW

File: cupy/cuda/thrust.pyx [hunk 1]
```diff
@@ -12,45 +12,47 @@ from cupy.cuda cimport common
...
 
 
 ###############################################################################
 # Python interface
 ###############################################################################
 
-cpdef sort(dtype, size_t start, size_t num):
-    cdef void* ptr
-    cdef Py_ssize_t n
+cpdef sort(dtype, size_t start, size_t ndim, size_t shape):
```
Comment (okuta) [inline cupy/cuda/thrust.pyx]: How about uinsg `vector.vector[Py_ssize_t]& shape`?
Please see [this code](https://github.com/cupy/cupy/blob/master/cupy/core/internal.pyx#L17)

Comment (takagi) [inline cupy/cuda/thrust.pyx]: Fixed!

File: cupy/cuda/cupy_thrust.cu [hunk 2]
```diff
@@ -12,22 +13,56 @@ using namespace thrust;
  */
 
 template <typename T>
-void cupy::thrust::_sort(void *start, ptrdiff_t num) {
-    device_ptr<T> dp_first = device_pointer_cast((T *)start);
-    device_ptr<T> dp_last  = device_pointer_cast((T *)start + num);
-    stable_sort< device_ptr<T> >(dp_first, dp_last);
+void cupy::thrust::_sort(void *start, std::vector<ptrdiff_t>& shape) {
```
Comment (okuta) [inline cupy/cuda/cupy_thrust.cu]: Could you add `const` to shape? Because this vector is not changed.

Comment (takagi) [inline cupy/cuda/cupy_thrust.cu]: Makes sense. Fixed!

**Comment by okuta** [comment]
LGTM!

**Pull request merged**